src/diffusers/loaders.py

```diff
@@ -763,6 +764,19 @@ def load_textual_inversion(
                 f" `{self.load_textual_inversion.__name__}`"
             )
 
+        # Remove any existing hooks.
+        is_model_cpu_offload = False
+        is_sequential_cpu_offload = False
+        for _, component in self.components.items():
+            if isinstance(component, nn.Module):
+                if hasattr(component, "_hf_hook"):
+                    is_model_cpu_offload = isinstance(getattr(component, "_hf_hook"), CpuOffload)
+                    is_sequential_cpu_offload = isinstance(getattr(component, "_hf_hook"), AlignDevicesHook)
+                    logger.info(
+                        "Accelerate hooks detected. Since you have called `load_textual_inversion()`, the previous hooks will be first removed. Then the textual inversion parameters will be loaded and the hooks will be applied again."
+                    )
+                    remove_hook_from_module(component)
+
```

Comment:
Doesn't one of these two hooks styles hook into every sub module as well, so shouldn't one of the checks be recursive?

Reply:
@muellerzr to help here a bit.

Reply:
@muellerzr to help here a bit.

Reply:
I don't think the hooks have to recursively be removed here. To me that looks fine

Reply:
You should be able to do `remove_hook_from_module(component, recursive=True)`

CC @SunMarc too for a second glance :)

Reply:
But is it required here? Sorry for not making my comment clear.

Reply:
I guess trying to understand just what we're aiming to achieve (solid guess based on context, let me know if I'm accurate):

- Given a model that may be loaded in via `device_map="auto"` or some form of `device_map`
- We wish to remove the hooks when wanting to load the full model into memory/remove it from BMI (big model inference)
- With the potential of placing it back later

Is this accurate? Otherwise may need a bit more info/context I'm missing somehow

Reply:
From what I understood from the codebase, if we have `is_sequential_cpu_offload`, it means that the components were offloaded using `cpu_offload` which places recursively the hooks on each submodules. In the case of `is_model_cpu_offload`, we use `cpu_offload_from_hook` which place only one hook on the module, so that the entire model will be offloaded when another hook is triggered. 
I would then suggest using `remove_hook_from_module(component, recursive=True)` for the first case and `remove_hook_from_module(component, recursive=False)` for the second case if you don't want to just recursively remove all the hooks for both cases !